example.go

```diff
@@ -0,0 +1,376 @@
+	// 	os.Exit(1)
+	// }
+
+	fmt.Printf("Delete Topic\n")
+	err = deleteTopic(serviceAPI)
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		os.Exit(1)
+	}
+
+	fmt.Printf("List Topics\n")
+	err = listTopics(serviceAPI)
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		os.Exit(1)
+	}
+}
+
+func listTopics(serviceAPI *adminrestv1.AdminrestV1) error {
+	// Construct an instance of the ListTopicsOptions model
+	listTopicsOptionsModel := new(adminrestv1.ListTopicsOptions)
+
+	// Call ListTopics.
+	result, response, operationErr := serviceAPI.ListTopics(listTopicsOptionsModel)
+	if operationErr != nil {
+		return fmt.Errorf("Error Listing Topics" + operationErr.Error())
+	}
+
+	// Check the result.
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("Error Listing Topics: status %d\n", response.StatusCode)
+	}
+
+	// Loop and print topics.
+	for _, topicDetail := range result {
+		fmt.Printf("\tname: %s\n", *topicDetail.Name)
+	}
+	return nil
+} // func.end
+
+func topicDetails(serviceAPI *adminrestv1.AdminrestV1) error {
+	// Construct an instance of the GetTopicOptions model
+	getTopicOptionsModel := new(adminrestv1.GetTopicOptions)
+	getTopicOptionsModel.TopicName = core.StringPtr("test-topic")
+	getTopicOptionsModel.Headers = map[string]string{"x-custom-header": "x-custom-value"}
+
+	// Call List Topic Details.
+	result, response, operationErr := serviceAPI.GetTopic(getTopicOptionsModel)
+	if operationErr != nil {
+		return fmt.Errorf("Error Listing Topic Details" + operationErr.Error())
+	}
+
+	// Check the result.
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("Error Listing Topic Details: status %d\n", response.StatusCode)
+	}
+
+	// Print topics details.
+	fmt.Printf("\tname: \t\t\t%s\n", *result.Name)
+
+	// The number of partitions.
+	fmt.Printf("\tno of partitions: \t%d\n", *result.Partitions)
+
+	// The number of replication factor.
+	fmt.Printf("\treplication factor: \t%d\n", *result.ReplicationFactor)
+
+	// // The value of config property 'retention.ms'.
+	fmt.Printf("\tretention (ms): \t%d\n", *result.RetentionMs)
+
+	// // The value of config property 'cleanup.policy'.
+	fmt.Printf("\tcleanup policy: \t%s\n", *result.CleanupPolicy)
+
+	// Configs *TopicConfigs
+	fmt.Printf("\ttopic configs: \t\t%+v\n", *result.Configs)
+
+	// The replia assignment of the topic.
+	// ReplicaAssignments []ReplicaAssignment
+	for _, assignment := range result.ReplicaAssignments {
+		fmt.Printf("\tassignment:  \t\tid:%d,  \tbrokers: %+v\n", assignment.ID, assignment.Brokers)
+	}
+	return nil
+} // func.end
+
+func createTopic(serviceAPI *adminrestv1.AdminrestV1) error {
+	// Construct an instance of the createTopicOptionsModel.
+	createTopicOptionsModel := new(adminrestv1.CreateTopicOptions)
+	createTopicOptionsModel.Name = core.StringPtr("test-topic")
+	createTopicOptionsModel.Partitions = core.Int64Ptr(int64(26))
+	createTopicOptionsModel.PartitionCount = core.Int64Ptr(int64(1))
```

Comment:
Do we need both of these?

Reply:
went for 
	createTopicOptionsModel.PartitionCount = core.Int64Ptr(int64(1))